src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -8541,6 +8541,246 @@ public void RefFieldsConsideredManaged_Indirect()
             Assert.True(comp.GetTypeByMetadataName("StructWithIndirectRefField").IsManagedTypeNoUseSiteDiagnostics);
         }
 
+        [WorkItem(62618, "https://github.com/dotnet/roslyn/issues/62618")]
+        [Theory]
+        [CombinatorialData]
+        public void RefAssignValueScopeMismatch_01(
+            [CombinatorialValues(LanguageVersion.CSharp10, LanguageVersion.CSharp11)] LanguageVersion languageVersion,
+            bool useUnsafe)
+        {
+            string unsafeModifier = useUnsafe ? "unsafe" : "";
+            var source =
+$@"using System;
+class Program
+{{
+    static void Main()
+    {{
+        Span<int> s = new[] {{ 1 }};
+        M(ref s);
+    }}
+    {unsafeModifier} static void M(ref Span<int> s)
+    {{
+        Span<int> local = stackalloc int[] {{ 1 }};
+        ref Span<int> rL = ref local;
+        rL = ref s; // 1
```

Comment:
I was thinking that this assignment should be permissible if the target is `ref readonly`. That is because a subsequent assignment like `rL = local` would not be possible.

Reply:
> I was thinking that this assignment should be permissible if the target is `ref readonly`.

That's a good catch, thanks.

I agree, we could allow assignment to `ref readonly` if necessary. For now, I've added a test.